projects/core/src/anonymous-consents/services/anonymous-consents-state-persistence.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Observable, Subscription } from 'rxjs';
+import { StatePersistenceService } from '../../state/index';
+import { AnonymousConsentsService } from '../facade/index';
+import {
+  AnonymousConsentsState,
+  StateWithAnonymousConsents,
+} from '../store/index';
+import { getAnonymousConsentState } from '../store/selectors/feature.selector';
+
+/**
+ * Anonymous consents state synced to browser storage.
+ */
+export type SyncedAnonymousConsentsState = Partial<AnonymousConsentsState>;
+
+/**
+ * Responsible for saving the anonymous consents data in browser storage.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class AnonymousConsentsStatePersistenceService implements OnDestroy {
+  protected subscription = new Subscription();
+
+  constructor(
+    protected statePersistenceService: StatePersistenceService,
+    protected store: Store<StateWithAnonymousConsents>,
+    protected anonymousConsentsService: AnonymousConsentsService
+  ) {}
+
+  /**
+   * Identifier used for storage key.
+   */
+  protected key = 'anonymous-consents';
+
+  /**
+   * Initializes the synchronization between state and browser storage.
+   */
+  public initSync() {
+    this.subscription.add(
+      this.statePersistenceService.syncWithStorage({
+        key: this.key,
+        state$: this.getAuthState(),
+        onRead: (state) => this.onRead(state),
+      })
+    );
+  }
+
+  /**
+   * Gets and transforms state from different sources into the form that should
+   * be saved in storage.
+   */
+  protected getAuthState(): Observable<SyncedAnonymousConsentsState> {
+    return this.store.select(getAnonymousConsentState);
+  }
+
+  /**
+   * Function called on each browser storage read.
+   * Used to update state from browser -> state.
+   */
+  protected onRead(state: SyncedAnonymousConsentsState) {
+    if (state) {
+      if (state.templates) {
+        this.anonymousConsentsService.getTemplates();
```

Comment:
Why if we have templates we invoke their load instead of restoring them with `AnonymousConsentsActions.LoadAnonymousConsentTemplatesSuccess`?